src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Diagnostics/BindingDiagnostics.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+//
+// Description:
+//  Binding diagnostics API
+//
+
+using System.Collections.Generic;
+
+namespace System.Windows.Diagnostics
+{
+    /// <summary>
+    /// Provides a notification infrastructure for listening to binding failure events.
+    /// </summary>
+    /// <remarks>
+    /// This type supports the .NET Framework infrastructure and is not intended to be used directly
+    /// from application code.
+    /// </remarks>
+    public static class BindingDiagnostics
+    {
+        internal static bool IsEnabled { get; private set; }
+
+        private static event EventHandler<BindingFailedEventArgs> bindingFailed;
+        private static List<BindingFailedEventArgs> pendingEvents;
```

Comment:
Naming conventions issue

See https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md

We use `_camelCase` for internal and private fields and use readonly where possible. Prefix internal and private instance fields with`_`, static fields with `s_` and thread static fields with `t_`.

Reply:
Thanks